java/applicationtags.md

```diff
@@ -0,0 +1,134 @@
+# Application Tags
+
+## Table of Content
+
+* [Overview](#Overview)
+* [Required Tags](#Required-Tags)
+* [Optional Tags](#Optional-Tags)
+* [Example](#Example)
+* [What's next](#What's-next)
+
+## Overview
+Many Wavefront SDKs require you to specify _application tags_ that describe the architecture of your application as it is deployed. These tags are associated with the metrics and trace data sent from the instrumented microservices in your application. You specify a separate set of application tags for each microservice you instrument. Wavefront uses these tags to aggregate and filter data at different levels of granularity.
+
+Application tags and their values are encapsulated in an `ApplicationTags` object in your microservice’s code. Because the tags describe the application’s architecture as it is deployed, your code typically obtains values for the tags from a YAML configuration file, either provided by the SDK or through a custom mechanism implemented by your application.
+
+
+## Required tags
+**Required tags** enable you to drill down into the data for a particular service:
+
+| Tag    | Description    |
+| :------------- | :------------- |
+| `application`     | Name that identifies your Java application, for example: `OrderingApp`. All microservices in an application should share the same application name.      |
+| `service`      | Name that identifies the microservice within your application, for example: `inventory`. Each microservice should have its own service name.       |
+
+
+## Optional tags
+**Optional tags** enable you to use the physical topology of your application to further filter your data:
+
+|  Tag      | Description     |
+| :------------- | :------------- |
+| cluster`      | Name of a group of related hosts that serves as a cluster or region in which the application runs, for example: `us-west-2`. The default value is `none`.       |
+| `shard`      | Name of a subgroup of hosts within a cluster that serve as a partition, replica, shard, or mirror, for example: `secondary`. The default value is `none`.       |
+
+Optionally, you can add custom tags specific to your application in the form of a `Map` (see example below).
+
+## Example
+
+Creating an `ApplicationTags` instance:
+```java
+String application = "OrderingApp";
+String service = "inventory";
+String cluster = "us-west-2";
+String shard = "secondary";
+
+Map<String, String> customTags = new HashMap<String, String>() {{
+  put("location", "Oregon");
+  put("env", "Staging");
+}};
+
+ApplicationTags applicationTags = new ApplicationTags.Builder(application, service).
+    cluster(cluster).       // optional
+    shard(shard).           // optional
+    customTags(customTags). // optional
+    build();
+```
+
+## What's next
+Click on the Java SDK you were using to continue instrumenting your applications with Wavefront.
```

Comment:
...Click on the Java SDK you are using...? ...you want to use...? 

instrumenting your applications with Wavefront. => Not sure "with Wavefront" makes sense here. Could probably omit.